docs/securing_installation.md

```diff
@@ -0,0 +1,108 @@
+
+Helm and Tiller are designed to install, remove, and modify logical applications that can contain many service interacting together. As a result, often its usefulness involves cluster-wide operations, which in a multitenant cluster means that great care must be taken with access to a cluster-wide Tiller installation to prevent improper activity. 
+
+Specific users and teams -- developers, operators, system and network administrators -- will need their own portion of the cluster in which they can use Helm and Tiller without risking other portions of the cluster. This means using a Kubernetes cluster with RBAC enabled and Tiller configured to enforce them. For more information about using RBAC in Kubernetes, see [Using RBAC Authorization](rbac.md).
+
+#### Tiller and User Permissions
+Tiller in its current form (2.7.x) does not provide a way to map user credentials to specific permissions within Kubernetes. When Tiller is running inside of the cluster, it operates with the permissions of its service account. If no service account name is supplied to Tiller, it runs with the default service account. This means that all Tiller operations on that server are executed using the Tiller pod's credentials and permissions. 
+
+The properly limit what Tiller itself can do, the standard Kubernetes RBAC mechanisms must be attached to Tiller, including Roles and RoleBindings that place explicit limits on what things a Tiller instance can install, and where. 
+
+This situation may change in the future. While the community has several methods that might address this, at the moment performing actions using the rights of the client, instead of the rights of Tiller, is contingent upon the outcome of the Pod Identity Working Group, which has taken on the task of solving the problem in a general way. 
+
+
+### The Tiller gRPC Endpoint and TLS
+
+In the default installation the gRPC endpoint that Tiller offers is available inside the cluster (not external to the cluster) without authentication configuration applied. Without applying authentication, any process in the cluster can use the gRPC endpoint to perform operations inside the cluster. In a local or secured private cluster, this enables rapid usage and is normal. (From outside the cluster, Helm uses the normal authentication that Kubernetes supports to reach a gRPC endpoint.)
+
+Shared and production clusters -- for the most part -- should use Helm 2.3.0 at a minimum and configure TLS for the Tiller gRPC endpoints to ensure that within the cluster usage of the gRPC endpoint is only for the properly authenticated identity. Doing so enables any number of Tiller instances to be deployed in any number of namespaces and yet no unauthenticated usage of any gRPC endpoint is possible. Finally, usage of Helm with the `--tiller-tls-verify` option to enforce TLS in any communication with gRPC endpoints.
```

Comment:
The minimum should be Helm 2.7.2 as TLS was pretty much busted before that.  See the 2.7.2 release notes: https://github.com/kubernetes/helm/releases/tag/v2.7.2

Reply:
It should be clear that each instance of Tiller should have its own TLS configuration (including a separate CA).  
Tiller currently doesn't look inside the certificate presented but just verifies it is signed by the CA.  If the same CA is used across tiller instances it will allow users to talk to any of them.

Reply:
> It should be clear that each instance of Tiller should have its own TLS configuration (including a separate CA).

This has already been documented in the linked doc: https://github.com/kubernetes/helm/blob/master/docs/tiller_ssl.md

Reply:
> Finally, usage of Helm with the `--tiller-tls-verify` option to enforce TLS

The flag name is `--tls` btw, not `--tiller-tls-verify`. That's just for `helm init`. All other commands use `--tls` to enforce TLS communication, and optionally `--tls-verify` to verify the server hostname in the cert. :)

Reply:
Doesn't `helm --tls` draw files from `$(helm home)/{ca,cert,key}.pem`? How does that work if you are talking to multiple tillers?

Reply:
Yes it does. Helm users being able to communicate with more than one tiller instance is an extreme use case in my opinion. I have seen many use cases where there are a single cluster with multiple tiller instances, but not *one user* with access to multiple tiller instances in that case. In most of those use cases, the administrator has wanted to lock down a user/group to only be able to deploy using one tiller instance to one (or many) namespaces.

In the case where there IS someone who communicates with multiple tiller instances on a daily basis, they can either change HELM_HOME to another location (effectively profile switching), or use `--tls-ca-cert`, `--tls-cert` and `--tls-key` to point to a different location.

Reply:
I suspect as folks understand the role that tiller has in a multi-tenant-ish cluster we will see more folks using multiple tillers.  This document, @technosophos's blog post and @michelleN 's project (https://github.com/michelleN/helm-secure-tiller) will all help to encourage users to do just that.

If setting `HELM_HOME` is the way to go we should probably document that at some point.

Reply:
Also, this will become critical as users start using secure tiller across multiple clusters, right?  You *could* use the same TLS certs across multiple clusters (say a cluster per region and a tiller per namespace) but you probably don't want to share TLS certs across environments (say, prod vs. dev) and often times folks will be operating across those environments.

Reply:
@bacongobbler thanks for the clarification, I appreciate it. I can see both scenarios being important as Helm is a very flexible and because of that important tool. I'll think about how we continue to describe specific approaches to using it here.